clients/src/test/java/org/apache/kafka/common/security/ssl/SslFactoryTest.java

```diff
@@ -379,4 +431,49 @@ public void testCertificateEntriesValidation() throws Exception {
     private TestSslUtils.SslConfigsBuilder sslConfigsBuilder(Mode mode) {
         return new TestSslUtils.SslConfigsBuilder(mode).tlsProtocol(tlsProtocol);
     }
+
+    public static final class TestSslEngineFactory implements SslEngineFactory {
```

Comment:
We could probably move this into TestSslUtils instead of duplicating in two tests.

Reply:
Done